types/newman/index.d.ts

```diff
@@ -1,14 +1,161 @@
+//                 Graham McGregor <https://github.com/Graham42>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import { EventEmitter } from "events";
+import {
+    Collection,
+    CollectionDefinition,
+    VariableScope,
+    VariableScopeDefinition
+} from "postman-collection";
+
+export interface NewmanRunOptions {
+    /** A JSON / Collection / String representing the collection. */
+    collection: Collection | CollectionDefinition | string;
+    /** An environment JSON / file path for the current collection run. */
+    environment?: VariableScope | VariableScopeDefinition | string;
+    /** A globals JSON / file path for the current collection run. */
+    globals?: VariableScope | VariableScopeDefinition | string;
+    /**
+     * Specify the number of iterations to run on the collection. This is
+     * usually accompanied by providing a data file reference as
+     * iterationData
+     */
+    iterationCount?: number;
+    /**
+     * Path to the JSON or CSV file or URL to be used as data source when
+     * running multiple iterations on a collection.
+     */
+    iterationData?: any;
+    /**
+     * The name or ID of the folder (ItemGroup) in the collection which would
+     * be run instead of the entire collection.
+     */
+    folder?: string;
+    /**
+     * Specify the time (in milliseconds) to wait for the entire collection run
+     * to complete execution.
+     *
+     * Default value: Infinity
+     */
+    timeout?: number;
+    /**
+     * Specify the time (in milliseconds) to wait for requests to return a
+     * response.
+     *
+     * Default value: Infinity
+     */
+    timeoutRequest?: number;
+    /**
+     * Specify the time (in milliseconds) to wait for scripts to return a
+     * response.
+     *
+     * Default value: Infinity
+     */
+    timeoutScript?: number;
+    /**
+     * Specify the time (in milliseconds) to wait for between subsequent
+     * requests.
+     *
+     * Default value: 0
+     */
+    delayRequest?: number;
+    /**
+     * This specifies whether newman would automatically follow 3xx responses
+     * from servers.
+     *
+     * Default value: false
+     */
+    ignoreRedirects?: boolean;
+    /**
+     * Disables SSL verification checks and allows self-signed SSL certificates.
+     *
+     * Default value: false
+     */
+    insecure?: boolean;
+    /**
+     * Specify whether or not to stop a collection run on encountering the
+     * first test script error.
+     *
+     * "folder" allows you to skip the entire collection run in case an invalid
+     * folder was specified using the `folder` option or an error was
+     * encountered in general.
+     *
+     * "failure" would gracefully stop a collection run after completing the
+     * current test script.
+     *
+     * Default value: false
+     */
+    bail?: boolean | ["folder"] | ["failure"];
```

Comment:
Is the difference between `"folder"` and `["folder"]` important in this case?

Reply:
Yup https://github.com/postmanlabs/newman/blob/release/3.x/lib/run/index.js#L101

Reply:
Thanks for confirming!